### PR TITLE
Catch compile errors, missing UPSTREAM commits

### PR DESCRIPTION
**Pull request by liggitt**

### CODE REVIEW

**Comment by liggitt** [comment]
@smarterclayton for `make verify` changes
@ironcladlou for commit checker change

**Comment by liggitt** [comment]
[test]

**Comment by ironcladlou** [comment]
Commit checker changes LGTM.

File: Makefile [hunk 1]
```diff
@@ -45,6 +45,10 @@ check: | build verify
 # Example:
 #   make verify
 verify: build
+	# Verify extended tests compile
+	$(MAKE) build WHAT=test/extended/extended.test
```
Comment (smarterclayton) [inline Makefile]: hack/build-go.sh test/extended/extended.test test/extended/networking/extended.test is preferred here

**Comment by bparees** [comment]
pretty sure k8s extended test compilation is broken right now, so +1.

**Comment by liggitt** [comment]
[merge]

**Pull request merged**